### PR TITLE
[ConstraintSystem] Allow fixing missing conformance failures for `Void` and uninhabited types.

### PR DESCRIPTION
**Pull request by hborla**

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]
Great!

File: test/Constraints/rdar44770297.swift [hunk 1]
```diff
@@ -4,8 +4,8 @@ protocol P {
   associatedtype A
 }
 
-func foo<T: P>(_: () throws -> T) -> T.A? { // expected-note {{in call to function 'foo'}}
+func foo<T: P>(_: () throws -> T) -> T.A? { // expected-note {{where 'T' = 'Never'}}
   fatalError()
 }
 
-let _ = foo() {fatalError()} & nil // expected-error {{generic parameter 'T' could not be inferred}}
+let _ = foo() {fatalError()} & nil // expected-error {{global function 'foo' requires that 'Never' conform to 'P'}}
```
Comment (slavapestov) [inline test/Constraints/rdar44770297.swift]: This is a clear improvement!

However I can't help but wonder about 'Void'. The diagnostic would say "global function 'foo' requires that 'Void' conform to 'P'", which is a true statement, but 'Void' cannot _ever_ conform to 'P'. Same if it's a function type or a metatype. As a follow-up you might want to add some test cases for non-nominal types and see if the diagnostic could be tailored a bit.

**Review by slavapestov — COMMENTED** [review]
I have an optional suggestion -- feel free to ignore.

**Comment by xedin** [comment]
I think https://github.com/apple/swift/pull/27133/files should have fixed the crash here...

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Pull request merged**